### PR TITLE
Revert testing workarounds

### PR DESCRIPTION
**Pull request by lieut-data**
#### Summary
Revert a series of debugging commits that have been made obsolete by running the product tests in isolation. We'll revisit all of this testing infrastructure more holistically.

#### Ticket Link
None, this is fallout of a monorepo.

#### Release Note
```release-note
NONE
```

### CODE REVIEW

File: server/channels/app/app_test.go [hunk 1]
```diff
@@ -37,24 +38,36 @@ func init() {
...
+	mockStore.On("Post").Return(&mockPostStore)
+	mockStore.On("System").Return(&mockSystemStore)
+	mockStore.On("License").Return(&mockLicenseStore)
+	mockStore.On("GetDBSchemaVersion").Return(1, nil)
+
 	prev := *th.App.Config().ServiceSettings.SiteURL
 
-	var called int32
 	th.App.AddConfigListener(func(old, current *model.Config) {
-		atomic.AddInt32(&called, 1)
```
Comment (wiggin77) [inline server/channels/app/app_test.go:47]: This one was to ensure some asserts are actually evaluated.  I think we should keep it.

File: server/channels/store/storetest/settings.go [hunk 2]
```diff
@@ -49,26 +48,12 @@ func log(message string) {
 	}
 }
 
-func getDefaultMysqlDSN() string {
```
Comment (wiggin77) [inline server/channels/store/storetest/settings.go:52]: I think we should keep these.  There is no scenario where the const default DSN works in CI.

**Review by wiggin77 — APPROVED** [review]
LGTM :+1:   Thanks for cleaning this up.

**Comment by lindy65** [comment]
> You don't have permissions to trigger this command.
It's only available for organization members.

I got this error via email too @lieut-data - am I not a member of the server repo?

**Comment by lieut-data** [comment]
@lindy65, I was going to raise that with devops as it might be indicative of some misconfiguration. Let me compose something.

**Pull request merged**